New admin table layout

tvdeyen opened the pull request:
The content of all table headers, state columns and action cells have text wrapping disabled now. For cases where we want to disble text wrapping in data cells we provide the `.no-wrap` class.

This also makes (very sensitive) use of the Bootstrap default table style. The main advantage is that Bootstrap uses rem (a relative unit) for table cell padding. As we plan to raise the body font size this is a huge win for us.

### Before

![table-before](https://cloud.githubusercontent.com/assets/42868/24084223/692758ea-0ce6-11e7-8992-773e3ac6261a.png)

### After

![table-after-13px](https://cloud.githubusercontent.com/assets/42868/24084225/6fd61582-0ce6-11e7-959b-562c5b47a699.png)

tvdeyen commented:
@Mandily another preparation for #1777

jhawthorn commented on backend/app/assets/stylesheets/spree/backend/shared/_tables.scss:
```diff
@@ -1,14 +1,20 @@
+    // If you want to disable this, add the .wrap-text class to the column.
+    &:not(.wrap-text) {
+      white-space: nowrap;
+    }
```
I'm not sure about this. I think our default should be to wrap, and to specify areas where wrapping is undesirable (like actions).

tvdeyen replied:
I think the default should be no wrap. Most of the time we display where little information in cells, mostly values (order numbers, email addresses, etc. You don't want to wrap these.

The only occasion where wrapping was useful (not even needed) was the image caption. It would be a much larger change set to add a no-wrapp class to all cells where we want this to happen. 

But this is just my preference. I'm also fine with having the opposite being the default. I would then vote for also making the header cells non wrapping. Nobody wants a 

"Created
at"

cell, right?

tvdeyen replied:
Took the opposite approach like suggested.

jhawthorn commented on backend/app/assets/stylesheets/spree/backend/shared/_tables.scss:
```diff
@@ -1,14 +1,20 @@
-  margin-bottom: 15px;
-  border-collapse: separate;
+  // Extend all tables with Bootstrap's default table style
+  @extend .table;
```
This isn't worse than what was here before (bootstrap's `.table` sets pretty much the same CSS as we already had here). It would be ideal if we were to move away from styling all `table` elements like this and instead required a class. That way we avoid needing to un-style the styles when we want a table to be "normal".

tvdeyen replied:
What would a "normal" table look like? Browser default? I always thought this is our default table style. But I understand your concern. Im also fine with adding a table class to all tables. The change set will be much larger then, though.

jhawthorn replied:
We can deal with this later. It's not a new issue and shouldn't block this PR.

tvdeyen commented:
@jhawthorn I updated the commit and took a more conservative approach. Only table headers, the `.actions` cell and states have text wrapping disabled. Added a `.no-wrap` class, so we can decide which cells we want to have text wrapping disabled as well.
